src/record/record.js

```diff
@@ -171,7 +171,7 @@ Record.prototype.set = function (pathOrData, dataOrNil) {
   let path = arguments.length === 1 ? undefined : pathOrData
   let data = arguments.length === 1 ? pathOrData : dataOrNil
 
-  if (path === undefined && (typeof data !== 'object' || Array.isArray(data) || data === null)) {
+  if (path === undefined && (!utils.isPlainObject(data) || Array.isArray(data) || data === null)) {
```

Comment:
Isn’t the isplainobject test enough here?

Reply:
lol

Reply:
didn't see the forest because of all the trees type of issue